tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/ConfigController.cs

```diff
@@ -0,0 +1,163 @@
+            IMatterCenterServiceFunctions matterCenterServiceFunctions,
+            IConfigRepository configRepository,
+            IHostingEnvironment hostingEnvironment
+            )
+        {
+            this.errorSettings = errorSettings.CurrentValue;
+            this.matterCenterServiceFunctions = matterCenterServiceFunctions;
+            this.configRepository = configRepository;
+            this.generalSettings = generalSettings.CurrentValue;
+            this.hostingEnvironment = hostingEnvironment;
+        }
+
+        /// <summary>
+        /// Returns all the entries for Configuring the UI
+        /// </summary>
+        /// <param name="configRequest">Request object for POST</param>   
+        [HttpPost("Get")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        public async Task<IActionResult> Get([FromBody] DynamicTableEntity configRequest)
+        {
+            string result = string.Empty;
+
+            try
+            {
+                #region Error Checking                
+                ErrorResponse errorResponse = null;
+
+                #endregion
+                var configResultsVM = await configRepository.GetConfigurationsAsync(configRequest);
+
+                createConfig(configResultsVM);
+                return matterCenterServiceFunctions.ServiceResponse(configResultsVM, (int)HttpStatusCode.OK);
+
+            }
+            catch (Exception exception)
+            {
+                customLogger.LogError(exception, MethodBase.GetCurrentMethod().DeclaringType.Name, MethodBase.GetCurrentMethod().Name, logTables.SPOLogTable);
+                throw;
+            }
+        }
+
+
+        private void createConfig(List<DynamicTableEntity> configs)
+        {
+            var configPath = Path.Combine(hostingEnvironment.WebRootPath, "app/uiconfig.js");
+            if (System.IO.File.Exists(configPath))
+                System.IO.File.Delete(configPath);
+
+            var configFile = System.IO.File.Open(configPath, FileMode.Create);
+            var configWriter = new StreamWriter(configFile, Encoding.UTF8);
+
+            List<string> configGroup = new List<string>();
+            EntityProperty key;
+            EntityProperty value;
+
+            foreach (DynamicTableEntity dt in configs)
+            {
+                bool hasKey = dt.Properties.TryGetValue("ConfigGroup", out value);
+                if (hasKey)
+                {
+                    if (!configGroup.Contains(value.StringValue))
+                    {
+                        configGroup.Add(value.StringValue);
+                    }
+                }
+            }
+
+            int groupCount = configGroup.Count;
+            int count = configs.Count;
+
+            int groups = 0;
+            configWriter.WriteLine("var uiconfigs = {");
+            foreach (string str in configGroup)
+            {
+                groups++;
+                configWriter.WriteLine("\"" + str + "\":  {");

```

Comment:
Check whether you can use Newtonsoft.Json to create this config.js file, instead of using this. hardcoded string literrals. See tha taxonomy.cs file in the repository project - C:\projects\mc2\tree\master\cloud\src\solution\Microsoft.Legal.MatterCenter.Service\Microsoft.Legal.MatterCenter.Repository\SharePointInfo